contentcuration/contentcuration/utils/parser.py

```diff
@@ -0,0 +1,68 @@
+"""
+--- NUMBER REGULAR EXPRESSIONS ---
+VALID_NUMBER: [MIXED_NUMBER | FRACTION | DECIMAL | PERCENTAGE | INTEGER]
+DECIMAL: INTEGER.UNFORMATTED_INT
+PERCENTAGE: [DECIMAL | MIXED_NUMBER | FRACTION]%
+MIXED_NUMBER: INTEGER FRACTION
+FRACTION: INTEGER/NON_ZERO_INT
+NON_ZERO_INT: SIGN[NON_ZERO_DIGIT]([DIGIT]{0,2}(,DIGIT{3})*|[UNFORMATTED_INT])?
+INTEGER: SIGN[FORMATTED_INT | UNFORMATTED_INT]
+FORMATTED_INT: [DIGIT]{1,3}[,DIGIT{3}]+
+UNFORMATTED_INT: [DIGIT]*
+DIGIT: [0-9]
+NON_ZERO_DIGIT: [1-9]
+SIGN: -{0,1}
+EXPONENT: [DECIMAL | INTEGER]e+{0,1}[INTEGER]
+
+"""
+import re
+
+SIGN = r'-?'
+DIGIT = r'[0-9]'
+NON_ZERO_DIGIT = r'[1-9]'
+UNFORMATTED_INT = re.compile("{digit}*".format(digit=DIGIT))
+FORMATTED_INT = re.compile("{digit}{{1,3}}(?:,{digit}{{3}})+".format(digit=DIGIT))
+INTEGER = re.compile("({sign}(?:{formatted}|{unformatted}))".format(sign=SIGN, formatted=FORMATTED_INT.pattern, unformatted=UNFORMATTED_INT.pattern))
+NON_ZERO_INT = re.compile("({sign}{non_zero}(?:{digit}{{0,2}}(?:,{digit}{{3}})+|{unformatted})?)".format(sign=SIGN, non_zero=NON_ZERO_DIGIT, digit=DIGIT, unformatted=UNFORMATTED_INT.pattern))
+FRACTION = re.compile("({integer}/{non_zero})".format(integer=INTEGER.pattern, non_zero=NON_ZERO_INT.pattern))
+MIXED_NUMBER = re.compile("({integer}) +({fraction})".format(integer=INTEGER.pattern, fraction=FRACTION.pattern))
+DECIMAL = re.compile("({integer}\.{unformatted})".format(integer=INTEGER.pattern, unformatted=UNFORMATTED_INT.pattern))
+VALID_NUMBER = re.compile("({decimal}|{mixed_number}|{fraction}|{integer})".format(decimal=DECIMAL.pattern, mixed_number=MIXED_NUMBER.pattern, fraction=FRACTION.pattern, integer=INTEGER.pattern))
+PERCENTAGE = re.compile("({num})%".format(num=VALID_NUMBER.pattern))
+EXPONENT = re.compile("((?:{decimal}|{integer})e\+?{integer})".format(decimal=DECIMAL.pattern, integer=INTEGER.pattern))
+
+def extract_value(text):
```

Comment:
I think some tests on at least the Python parser would increase confidence here. Mainly just catching the use cases you described to me previously: `['1,000', '1 1/3', '1e10', '2.0e+10', '45/3', '3.333', '-1 1/2', '2.1e-10']`. And a few invalid values to check they convert to None.